src/org/thoughtcrime/securesms/qr/QrActivity.java

```diff
@@ -125,10 +171,54 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
         }
     }
 
+    @Override
+    public void onActivityResult(int reqCode, int resultCode, final Intent data) {
+        super.onActivityResult(reqCode, resultCode, data);
+
+        if (resultCode != Activity.RESULT_OK)
+            return;
+
+        switch (reqCode) {
+            case REQUEST_CODE_IMAGE:
+                Uri uri  = (data != null ? data.getData() : null);
+                if (uri != null) {
+                    try {
+                        InputStream inputStream = getContentResolver().openInputStream(uri);
+                        Bitmap bitmap = BitmapFactory.decodeStream(inputStream);
+                        if (bitmap == null) {
+                            Log.e(TAG, "uri is not a bitmap: " + uri.toString());
+                            return;
+                        }
+                        int width = bitmap.getWidth(), height = bitmap.getHeight();
+                        int[] pixels = new int[width * height];
+                        bitmap.getPixels(pixels, 0, width, 0, 0, width, height);
+                        bitmap.recycle();
+                        bitmap = null;
+                        RGBLuminanceSource source = new RGBLuminanceSource(width, height, pixels);
+                        BinaryBitmap bBitmap = new BinaryBitmap(new HybridBinarizer(source));
+                        MultiFormatReader reader = new MultiFormatReader();
+                        try {
+                            Result result = reader.decode(bBitmap);
```

Comment:
currently Desktop also fails silently :/

Reply:
there is string `qrscan_failed` ("QR code could not be decoded") that could be used instead of adding a new string, what do you think? I think it is fine, btw, it seems that string is currently unused, good opportunity to recycle all that translation efforts :D

Reply:
I showed a toast instead of a dialog

Reply:
btw, the error is already captured, just that only a log entry was written without user feedback